src/resources-controller_impl.cxx

```diff
@@ -121,15 +123,17 @@ void
 ResourcesControllerImpl::add_resource_group_done (
   ResourceGroupController::DoneType done_type )
 {
-  m_resource_group_controller->hide();
```

Comment:
Removing this line wasn't necessary.

Reply:
void
ResourcesControllerImpl::add_resource_done (
 ResourceController::DoneType done_type )
{
 m_resource_controller->hide();
 if ( done_type == ResourceController::DoneType::CANCEL )
   {
     return;
   }
 m_resource_controller.reset();
 auto& res = m_resource;
 auto& group = *m_rm->get_resource_group_mutable ( group_id );
 group.add_resource ( res );
 m_view->add_resource ( group_index, res.id(), res.name(), res.cost(), res.description() );
}

this was the condition of code before I made any changes.

the problem with this code is:

it hides the dialogue box regardless of the done_type. If CANCEL is done_type, the dialogue box is hidden and even when DONE is done_type the dialogue box is hidden.

why is this a problem:
it is a problem because if CANCEL is done_type then of course we want to hide the dialogue box.

but we don't want to do so if the done_type is DONE:

Because, if done_type is DONE that means we have to check the name of resource to satisfy the constraints (unique and not null) set by model.

if it satisfies the constraint then we would hide the dialogue box with *->hide(); and the reset the view by *.reset();

but if it doesn't then we neither want the dialogue box to hide nor do we want the view to reset.
we just keep the dialogue box as it is, so that user would change the name in a way so as to satisfy the said constraints.

so that is why the above code became this:

void
ResourcesControllerImpl::add_resource_done (
  ResourceController::DoneType done_type )
{
  if ( done_type == ResourceController::DoneType::CANCEL )
    {
      m_resource_controller->hide();
      m_resource_controller.reset();
      return;
    }
  auto& res = m_resource;
  auto& group = *m_rm->get_resource_group_mutable ( group_id );
  if ( group.add_resource ( res ) )
    {
      m_resource_controller->hide();
      m_resource_controller.reset();
      m_view->add_resource ( group_index, res.id(), res.name(), res.cost(), res.description() );
    }
}

Same goes for resource_group as well.

Reply:
we can of course remove the extra reset statement in the CANCEL if block.